frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -153,10 +153,13 @@ export default class FirebaseInteractor {
     return { id, currentIndex, words: actualWords, firebaseId: assessment.id };
   }
 
-  async updateAssessment(id: string, responses: AssessmentResult[]) {
+  async updateAssessment(id: string, responses: AssessmentResult[], currentIdx: number) {
     await Promise.all(responses.map(async (response) => {
       return await this.db.collection("assessments").doc(id).collection("results").doc(response.word).set({correct: response.correct});
```

Comment:
i guess this is from before but is there a reason why this is set but not update?

Reply:
This one is set since there is a possibility that the actual doc does not exist. Update fails if the doc does not exist. Generally when creating something new you should use `set` and when updating use `update`